plugins/userhomepage/UserHomePagePlugin.php

```diff
@@ -0,0 +1,187 @@
+                'LabelCode' => 'Home Page',
+                "Description" => sprintf(
+                    t("Choose the page you would like to see when you visit visit: %s."),
+                    "<strong><a href='$homeUrl'>$homeUrl</a></strong>"
+                ),
+                'Control' => 'RadioList',
+                'Items' => [
+                    '/categories' => '/categories',
+                    '/discussions' => '/discussions',
+                ],
+                'Options' => ['Default' => $this->getDefaultHomepage()],
+            ],
+        ];
+
+        $userPref = $this->getUserHomepagePreference() ?: $this->getDefaultHomepage();
+        if ($userPref !== null) {
+            $sender->Form->setData(['homepage' => $userPref]);
+        }
+
+        // Handle the form post back.
+        if ($sender->Form->authenticatedPostBack()) {
+            $values = $sender->Form->formValues();
+            $newHomepagePref = $values['homepage'] ?? null;
+            if ($newHomepagePref !== null) {
+                $this->userMetaModel->setUserMeta($this->session->UserID, self::PREF_KEY, $newHomepagePref);
+            }
+        }
+        $sender->setData('formTemplate', $formTemplate);
+        $sender->setData('form', $sender->Form);
+        $sender->render('homepage', '', 'plugins/userhomepage');
+    }
+
+    /**
+     * Add "Home Page Settings" to profile edit mode side menu.
+     *
+     * @param \ProfileController $sender
+     */
+    public function profileController_afterAddSideMenu_handler(\ProfileController $sender) {
+        if (!$this->canEditUsersPreference($sender)) {
+            return;
+        }
+
+        /** @var \MenuModule $sideMenu */
+        $sideMenu = $sender->EventArguments['SideMenu'];
+        $sideMenu->addLink(
+            'Options',
+            t('Home Page Settings'),
+            userUrl($sender->User, '', 'homepage')
+        );
+    }
+
+    /**
+     * Add "Home Page Settings" to Profile Edit button group for user's that can't get navigate to edit profile.
+     *
+     * @param \ProfileController $sender
+     * @param array $args
+     */
+    public function profileController_beforeProfileOptions_handler(\ProfileController $sender, array $args) {
+        $canEditProfiles = checkPermission('Garden.Users.Edit') || checkPermission('Moderation.Profiles.Edit');
+        if ($this->canEditUsersPreference($sender) && !$canEditProfiles) {
+            $args['ProfileOptions'][] = [
+                'Text' => t('Home Page Settings'),
+                'Url' => userUrl($sender->User, '', 'homepage')
+            ];
+        }
+    }
+
+    /**
+     * Determine if the sessioned user can edit the home page settings of a particular user.
+     *
+     * @param \ProfileController $profileController
+     * @return bool
+     */
+    private function canEditUsersPreference(\ProfileController $profileController): bool {
+        $userID = $profileController->User->UserID;
```

Comment:
`$profileController->User` can be either an object or array here, this is throwing an error:

![Screen Shot 2020-01-17 at 2 14 52 PM](https://user-images.githubusercontent.com/31521020/72639555-ce91d300-3933-11ea-9d75-69f8873fe66b.png)